packages/seo-integration/src/analyze.js

```diff
@@ -0,0 +1,144 @@
+import { Paper } from "yoastseo";
+
+/**
+ * Creates a paper based on the given data, keyphrase and configuration.
+ *
+ * @param {Object} data The paper data and metadata, e.g. content, title, date, slug.
+ * @param {Object} keyphrase The keyphrase to use for the analysis.
+ * @param {Object} configuration Extra configuration.
+ *
+ * @returns {Paper} A paper that can be analyzed using the web worker.
+ */
+function createPaper( data, keyphrase, configuration ) {
+	return new Paper(
+		data.content,
+		{
+			// Keyphrase data.
+			keyword: keyphrase.keyphrase,
+			synonyms: keyphrase.synonyms,
+			// General data and metadata.
+			description: data.metaDescription,
+			title: data.seoTitle,
+			titleWidth: data.seoTitleWidth,
+			permalink: data.slug,
+			date: data.date,
+			// Configuration data.
+			locale: configuration.locale,
+		},
+	);
+}
+
+/**
+ * Adapts the object with related keyphrases to a a structure
+ * that the analysis web worker is able to consume.
+ *
+ * @param {Object} relatedKeyphrases The related keyphrases.
+ *
+ * @returns {Object} The transformed related keyphrases.
+ */
+function adaptRelatedKeyphrases( relatedKeyphrases ) {
```

Comment:
We use the term `transform` instead of `adapt` in similar cases. Is that something you considered?